velox/expression/FunctionSignature.cpp

```diff
@@ -91,6 +91,12 @@ TypeSignature parseTypeSignature(const std::string& signature) {
   boost::algorithm::trim(token);
   nestedTypes.emplace_back(parseTypeSignature(token));
 
+  if (baseType == "SHORT_DECIMAL") {
```

Comment:
Not possible to avoid this hardcoding since decimal type parsing is different from the other types.
The difference comes from extracting the precision and scale variables.

Reply:
In my document, I proposed using flex/bison for this as well.